linkerd/app/inbound/src/target.rs

```diff
@@ -351,14 +351,14 @@ impl<A> svc::stack::RecognizeRoute<http::Request<A>> for RequestTarget {
 
 // === impl Logical ===
 
-impl From<(Option<profiles::Receiver>, Target)> for Logical {
-    fn from((profiles, target): (Option<profiles::Receiver>, Target)) -> Self {
+impl From<(profiles::Receiver, Target)> for Logical {
+    fn from((profiles, target): (profiles::Receiver, Target)) -> Self {
         Self { profiles, target }
     }
 }
 
 impl Param<Option<profiles::Receiver>> for Logical {
     fn param(&self) -> Option<profiles::Receiver> {
-        self.profiles.clone()
+        Some(self.profiles.clone())
```

Comment:
Seems like we could update the router to require a profile now?

Reply:
That's what ##967 does :) It's in a separate branch because that also depends on the outbound changes.